### PR TITLE
Add hopannotation1 schema to legacy ndt/traceroute table schema

### PR DESCRIPTION
**Pull request by cristinaleonr**
Add hopannotation1 annotations to legacy ndt/traceroute table in preparation for annotation export process.

[Query](https://pantheon.corp.google.com/bigquery?project=mlab-sandbox&ws=!1m5!1m4!1m3!1smlab-sandbox!2sbquxjob_5bc5cf08_17d4969ad4c!3sUS&j=bq:US:bquxjob_5bc5cf08_17d4969ad4c&page=queryresults) to see sample output:
`SELECT *
FROM
  `mlab-sandbox.base_tables.traceroute`
WHERE _PARTITIONTIME = "2020-10-23 00:00:00 UTC" AND uuid = "ndt-ft4fh_1602604312_0000000000096C60";`

This change is [Reviewable](https://reviewable.io/reviews/m-lab/etl/1030)

### CODE REVIEW

File: cmd/update-schema/update.go [hunk 1]
```diff
@@ -248,12 +248,12 @@ func updateStandardTables(project string) int {
 		errCount++
 	}
 
-	if err := CreateOrUpdatePCAPRow(project, "tmp_ndt", "pcap"); err != nil {
-		errCount++
-	}
-	if err := CreateOrUpdatePCAPRow(project, "raw_ndt", "pcap"); err != nil {
-		errCount++
-	}
+	// if err := CreateOrUpdatePCAPRow(project, "tmp_ndt", "pcap"); err != nil {
```
Comment (cristinaleonr) [inline cmd/update-schema/update.go:251]: This needs to be uncommented to merge.

**Review by gfr10598 — CHANGES_REQUESTED** [review]
> Reviewed 4 of 8 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/m-lab/etl/1030)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @cristinaleonr)

___
*[parser/parser.go, line 81 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-MpgfYMeEsLok8-CvCTS:-MpgfYMeEsLok8-CvCTT:b8hx0q) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/parser.go#L81)):*
> ```Go
> }
> 
> // GetHopID creates a unique identifier to join Hop Annotations
> ```

to join synthetic Hop Annotations ...
___
*[parser/parser.go, line 83 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpge1_s9_upnephXNq0:-Mpge1_s9_upnephXNq1:b-4xp03a) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/parser.go#L83)):*
> ```Go
> // GetHopID creates a unique identifier to join Hop Annotations
> // with traceroute datasets.
> func GetHopID(cycleStartTime float64, hostname string, address string) string {
> ```

Does this code or something close already exist in traceroute-caller?  Or is this considered a synthetic ID?

Maybe call this GetExporterHopID, or GetSyntheticHopID?
___
*[parser/pt.go, line 281 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpgf2QXDGprNpi3P1LU:-Mpgf2QXDGprNpi3P1LV:bj6kusf) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/pt.go#L281)):*
> ```Go
> 		hopAnn := &hopannotation.HopAnnotation1{ID: hopID, Timestamp: time.Unix(int64(cycleStart.Start_time), 0).UTC()}
> 		hops = append(hops, schema.ScamperHop{
> 			Source: schema.HopIP{IP: oneNode.Addr, Hostname: oneNode.Name,
> ```

Can you change oneNode.Name to Hostname, or something that clearly identifies what the field means?
___
*[schema/pt_schema.go, line 24 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpgg1av61DvtcZVDzSR:-Mpgg1av61DvtcZVDzSS:b-36k4fx) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/schema/pt_schema.go#L24)):*
> ```Go
> 	Hostname       string                        `json:"hostname"`
> 	ASN            uint32                        `json:"asn,uint32"`
> 	HopAnnotation1 *hopannotation.HopAnnotation1 `json:"hopannotation1" bigquery:"hopannotation1"`
> ```

I noticed in pcap parser that you have the bigquery annotations, but not the json annotations.  So when running with json output, it doesn't change the field names.

Here, you probably want both json and bigquery annotations.  Otherwise, when running in legacy pipeline, you will get IP, City, etc. as field names in bigquery.  In any case, whichever one you want, you should probably have both json and bigquery annotations, so that it will be consistent when we migrate.

**Review by cristinaleonr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/m-lab/etl/1030)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @gfr10598)

___
*[parser/parser.go, line 81 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-MpgfYMeEsLok8-CvCTS:-MphTo482QclJ6PmTty5:b-hhbtjg) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/parser.go#L81)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

to join synthetic Hop Annotations ...
</blockquote></details>

This ID is also used to join real Hop Annotations with new traceroute datasets (`parser/scamper1.go` line 84) .
___
*[parser/parser.go, line 83 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpge1_s9_upnephXNq0:-MphWqKv1NOPPujzXOai:b-lt2wwa) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/parser.go#L83)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Does this code or something close already exist in traceroute-caller?  Or is this considered a synthetic ID?

Maybe call this GetExporterHopID, or GetSyntheticHopID?

</blockquote></details>

Something similar does exist in traceroute-caller. That said, the logic is spread across two files:

https://github.com/m-lab/traceroute-caller/blob/773bb092b18589d2fee20418ed1fa9aa6c5850cc/triggertrace/triggertrace.go#L147
https://github.com/m-lab/traceroute-caller/blob/773bb092b18589d2fee20418ed1fa9aa6c5850cc/hopannotation/hopannotation.go#L235
https://github.com/m-lab/traceroute-caller/blob/773bb092b18589d2fee20418ed1fa9aa6c5850cc/hopannotation/hopannotation.go#L237
___
*[parser/pt.go, line 281 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpgf2QXDGprNpi3P1LU:-MphYUsP5edJGKBhu0-s:b-sbuiid) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/pt.go#L281)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Can you change oneNode.Name to Hostname, or something that clearly identifies what the field means?
</blockquote></details>

Changed.
___
*[schema/pt_schema.go, line 24 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpgg1av61DvtcZVDzSR:-Mphb-pD-RJh-t67RIB_:b-yc5cjk) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/schema/pt_schema.go#L24)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

I noticed in pcap parser that you have the bigquery annotations, but not the json annotations.  So when running with json output, it doesn't change the field names.

Here, you probably want both json and bigquery annotations.  Otherwise, when running in legacy pipeline, you will get IP, City, etc. as field names in bigquery.  In any case, whichever one you want, you should probably have both json and bigquery annotations, so that it will be consistent when we migrate.
</blockquote></details>

Thanks for the note on the missing json tags for PCAP.

I'd added both tags to this `HopAnnotation1` field, which is the one I'm introducing with this PR :). I added the additional bigquery tags to the other fields in the struct, although these have already existed for years, so I just copied the name they have in the table today. 

To clarify, this is the legacy schema, which will be deleted when we migrate.

**Review by gfr10598 — APPROVED** [review]
:lgtm: after responses are addressed.

> Reviewed 3 of 8 files at r1, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/m-lab/etl/1030#-:-MpiXfB08tsHjLtX1_4k:bbne4oy)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @cristinaleonr)

___
*[parser/parser.go, line 81 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-MpgfYMeEsLok8-CvCTS:-MpiWh3w6VyESNk1AZIP:b-vzkme8) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/parser.go#L81)):*
<details><summary><i>Previously, cristinaleonr (Cristina Leon) wrote…</i></summary><blockquote>

This ID is also used to join real Hop Annotations with new traceroute datasets (`parser/scamper1.go` line 84) .
</blockquote></details>

In that case (and in light of next response below) best to add a comment here cross referencing the corresponding code in traceroute-caller.  Maybe also add an issue in traceroute-caller suggesting a cross-reference to this code.  Use permalinks?
___
*[schema/pt_schema.go, line 24 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpgg1av61DvtcZVDzSR:-MpiX7WzC5Ym3A3q6LqV:b-bbi6hm) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/schema/pt_schema.go#L24)):*
<details><summary><i>Previously, cristinaleonr (Cristina Leon) wrote…</i></summary><blockquote>

Thanks for the note on the missing json tags for PCAP.

I'd added both tags to this `HopAnnotation1` field, which is the one I'm introducing with this PR :). I added the additional bigquery tags to the other fields in the struct, although these have already existed for years, so I just copied the name they have in the table today. 

To clarify, this is the legacy schema, which will be deleted when we migrate. 
</blockquote></details>

Good.  Thanks for clarification.  Guess I should have noticed that those fields are pre-existing.

Now - for standard columns we are camel casing fields with structs, but naming top level structs with lower case?  Is that right?  Should we use that convention here?

**Review by cristinaleonr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/m-lab/etl/1030)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @gfr10598)

___
*[parser/parser.go, line 81 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-MpgfYMeEsLok8-CvCTS:-MplX0DQ-TX2Iu38NBUH:b-5dd6pg) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/parser/parser.go#L81)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

In that case (and in light of next response below) best to add a comment here cross referencing the corresponding code in traceroute-caller.  Maybe also add an issue in traceroute-caller suggesting a cross-reference to this code.  Use permalinks?
</blockquote></details>

I added a reference here to the traceroute-code. And I added a new [issue](https://github.com/m-lab/traceroute-caller/issues/129) to traceroute-caller 
___
*[schema/pt_schema.go, line 24 at r3](https://reviewable.io/reviews/m-lab/etl/1030#-Mpgg1av61DvtcZVDzSR:-MplXCJn5jsUlrnYkYiD:b3c8k5u) ([raw file](https://github.com/m-lab/etl/blob/9113a4f5fc94aa9121ebc3c1523877474f8fe6b1/schema/pt_schema.go#L24)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Good.  Thanks for clarification.  Guess I should have noticed that those fields are pre-existing.

Now - for standard columns we are camel casing fields with structs, but naming top level structs with lower case?  Is that right?  Should we use that convention here?
</blockquote></details>

Yes, I believe so. I'd made it lowercase for the sake of uniform names, but it makes more sense to follow standard columns. I have changed the bigquery tag to camel case.

**Pull request merged**